membrane_curvature/base.py

```diff
@@ -179,8 +183,20 @@ def _single_frame(self):
                                                                   y_range=self.y_range)
         self.results.mean[self._frame_index] = mean_curvature(self.results.z_surface[self._frame_index])
         self.results.gaussian[self._frame_index] = gaussian_curvature(self.results.z_surface[self._frame_index])
+        # Perform interpolation fo surface
+        if self.interpolation:
+            # Populate a slice with np.arrays of interpolated surface
+            self.results.interpolated_z_surface[self._frame_index] = surface_interpolation(
+                self.results.z_surface[self._frame_index])
 
     def _conclude(self):
         self.results.average_z_surface = np.nanmean(self.results.z_surface, axis=0)
         self.results.average_mean = np.nanmean(self.results.mean, axis=0)
         self.results.average_gaussian = np.nanmean(self.results.gaussian, axis=0)
+        if self.interpolation:
+            self.results.average_interpolated_z_surface = np.nanmean(
+                self.results.interpolated_z_surface[self._frame_index], axis=0)
```

Comment:
```suggestion
                self.results.interpolated_z_surface, axis=0)
```

And with the others too -- if you specify `frame_index`, you'll wind up only getting the mean of the last frame.

Reply:
Do you need `np.nanmean`? Do you expect `np.nan` values in your interpolated surfaces? If not, using `np.mean` will reveal errors in case there ever are `np.nan` values.